File: backend/app/models/note_tag.py
from sqlalchemy import ForeignKey, String, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column

from ..database import Base


class NoteTag(Base):
    __tablename__ = "note_tags"
    __table_args__ = (UniqueConstraint("family_id", "name", name="uq_note_tag_family_name"),)

    id: Mapped[int] = mapped_column(primary_key=True)
    family_id: Mapped[int] = mapped_column(
        ForeignKey("families.id", ondelete="CASCADE"), index=True
    )
    name: Mapped[str] = mapped_column(String(50))
    color: Mapped[str] = mapped_column(String(7), default="#6B7280")

File: flutter/lib/features/notes/domain/note_tag.dart
class NoteTag {
  final int id;
  final String name;
  final String color;

  const NoteTag({
    required this.id,
    required this.name,
    required this.color,
  });

  factory NoteTag.fromJson(Map<String, dynamic> json) {
    return NoteTag(
      id: json['id'] as int,
      name: json['name'] as String,
      color: json['color'] as String? ?? '#6B7280',
    );
  }

  Map<String, dynamic> toJson() => {
        'name': name,
        'color': color,
      };
}
